File: sample/src/main/java/com/airmap/airmapsdktest/MapDemoActivity.java
package com.airmap.airmapsdktest;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;

import com.airmap.airmapsdk.models.airspace.AirMapAirspaceAdvisoryStatus;
import com.airmap.airmapsdk.models.rules.AirMapRuleset;
import com.airmap.airmapsdk.ui.activities.MyLocationMapActivity;
import com.airmap.airmapsdk.ui.views.CustomViewPager;
import com.airmap.airmapsdk.util.Utils;

import java.util.List;

/**
 * Created by collin@airmap.com on 9/8/17.
 */

public class MapDemoActivity extends MyLocationMapActivity {

    private static final String TAG = "MapDemoActivity";

    private Toolbar toolbar;
    private TabLayout tabLayout;
    private MapPagerAdapter pagerAdapter;
    private CustomViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupViews();
    }

    private void setupViews() {
        toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle(getTitle());
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        pagerAdapter = new MapPagerAdapter(getSupportFragmentManager());

        viewPager = findViewById(R.id.view_pager);
        viewPager.setOffscreenPageLimit(3);
        viewPager.setPagingEnabled(false);
        viewPager.setAdapter(pagerAdapter);

        tabLayout = findViewById(R.id.tab_layout);
        tabLayout.setupWithViewPager(viewPager);
        tabLayout.setTabMode(TabLayout.MODE_FIXED);
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
    }

    public void setRulesets(List<AirMapRuleset> availableRulesets, List<AirMapRuleset> selectedRulesets) {
        getRulesetsFragment().setRulesets(availableRulesets, selectedRulesets);
    }

    public void selectRuleset(AirMapRuleset ruleset) {
        getMapFragment().onRulesetSelected(ruleset);
    }

    public void deselectRuleset(AirMapRuleset ruleset) {
        getMapFragment().onRulesetDeselected(ruleset);
    }

    public void switchSelectedRulesets(AirMapRuleset fromRuleset, AirMapRuleset toRuleset) {
        getMapFragment().onRulesetSwitched(fromRuleset, toRuleset);
    }

    public void setAdvisoryStatus(AirMapAirspaceAdvisoryStatus status) {
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (fragment instanceof AdvisoriesFragment) {
                ((AdvisoriesFragment) fragment).setAdvisoryStatus(status);
            }
        }
    }

    private MapFragment getMapFragment() {
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (fragment instanceof MapFragment) {
                return (MapFragment) fragment;
            }
        }

        return null;
    }

    private RulesetsFragment getRulesetsFragment() {
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (fragment instanceof RulesetsFragment) {
                return (RulesetsFragment) fragment;
            }
        }

        return null;
    }

    private class MapPagerAdapter extends FragmentPagerAdapter {

        public MapPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 0:
                    return MapFragment.newInstance();
                case 1:
                   return RulesetsFragment.newInstance();
                case 2:
                    return AdvisoriesFragment.newInstance();
            }

            return null;
        }

        @Override
        public String getPageTitle(int position) {
            switch (position) {
                case 0:
                    return "Map";
                case 1:
                    return "Rules";
                case 2:
                    return "Advisories";
            }

            return "";
        }

        @Override
        public int getCount() {
            return 3;
        }
    }
}

File: sample/src/main/java/com/airmap/sample/activities/MapDemoActivity.kt
package com.airmap.sample.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v7.widget.Toolbar

import com.airmap.airmapsdk.models.rules.AirMapRuleset
import com.airmap.airmapsdk.models.status.AirMapAirspaceStatus
import com.airmap.airmapsdk.ui.activities.MyLocationMapActivity
import com.airmap.airmapsdk.ui.views.AirMapMapView
import com.airmap.airmapsdk.ui.views.SwipeDisableableViewPager
import com.airmap.sample.R
import com.airmap.sample.fragments.AdvisoriesFragment
import com.airmap.sample.fragments.MapFragment
import com.airmap.sample.fragments.RulesetsFragment

class MapDemoActivity : MyLocationMapActivity() {

    companion object {
        @JvmStatic
        fun Context.MapDemoIntent() = Intent(this, MapDemoActivity::class.java)
    }

    private val toolbar by lazy { findViewById<Toolbar>(R.id.toolbar) }
    private val tabLayout by lazy { findViewById<TabLayout>(R.id.tab_layout) }
    private val viewPager by lazy { findViewById<SwipeDisableableViewPager>(R.id.view_pager) }

    private val mapFragment: MapFragment? get() = supportFragmentManager.fragments.filterIsInstance<MapFragment>().firstOrNull()
    private val rulesetsFragment: RulesetsFragment? get() = supportFragmentManager.fragments.filterIsInstance<RulesetsFragment>().firstOrNull()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupViews()
    }

    private fun setupViews() {
        toolbar.title = title
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)


        viewPager.setPagingEnabled(false)
        viewPager.offscreenPageLimit = 3
        viewPager.adapter = MapPagerAdapter(supportFragmentManager)

        tabLayout.tabMode = TabLayout.MODE_FIXED
        tabLayout.tabGravity = TabLayout.GRAVITY_FILL
        tabLayout.setupWithViewPager(viewPager)
    }

    fun setRulesets(availableRulesets: List<AirMapRuleset>, selectedRulesets: List<AirMapRuleset>) = rulesetsFragment?.setRulesets(availableRulesets, selectedRulesets)

    fun selectRuleset(ruleset: AirMapRuleset) = mapFragment?.onRulesetSelected(ruleset)

    fun deselectRuleset(ruleset: AirMapRuleset) = mapFragment?.onRulesetDeselected(ruleset)

    fun switchSelectedRulesets(fromRuleset: AirMapRuleset, toRuleset: AirMapRuleset) = mapFragment?.onRulesetSwitched(fromRuleset, toRuleset)

    fun setAdvisoryStatus(status: AirMapAirspaceStatus) = supportFragmentManager.fragments
            .filterIsInstance<AdvisoriesFragment>()
            .forEach { it.setAdvisoryStatus(status) }

    override fun getMapView(): AirMapMapView? = mapFragment?.mapView

    private inner class MapPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment = when (position) {
            0 -> MapFragment.newInstance()
            1 -> RulesetsFragment.newInstance()
            else -> AdvisoriesFragment.newInstance()
        }

        override fun getPageTitle(position: Int) = when (position) {
            0 -> "Map"
            1 -> "Rules"
            2 -> "Advisories"
            else -> ""
        }

        override fun getCount() = 3
    }
}
